homeassistant/components/sensor/sochain.py

```diff
@@ -0,0 +1,78 @@
+"""
+Support for watching multiple cryptocurrencies.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.sochain/
+"""
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['python-sochain-api==0.0.1']
+
+CONF_ADDRESS = 'address'
+CONF_NETWORK = 'network'
+CONF_ATTRIBUTION = "Data provided by chain.so"
+
+DEFAULT_NAME = 'Crypto Balance'
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ADDRESS): cv.string,
+    vol.Required(CONF_NETWORK): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the sochain sensors."""
+    address = config.get(CONF_ADDRESS)
+    network = config.get(CONF_NETWORK)
+    name = config.get(CONF_NAME)
+
+    add_devices([SochainSensor(name, network, address)], True)
```

Comment:
You should check before setting up the sensor if the given address is valid. Otherwise the users will end up with a non-functional platform in their setup.

Reply:
There really isn't a correct way to check an address besides looking it up.

Cryptocurrencies evolve over time and can change address format, and new networks can be added to sochain. (even if I just take what is supported right now, there would be 16 different formats for the 4 currencies and their testnet versions)

Putting address validation in here will require an update to this component every time any supported currency (or their test version) changes. It's generally not a good idea.

Is there another way to prevent the non-functional platform without doing local address validation?

Reply:
Would doing a lookup in the setup_platform (or async_setup_platform) and bailing out if I get an API failure be a good way of handling this?